From df1306a96127e91ff2d513a0a67345baaf61d113 Mon Sep 17 00:00:00 2001
From: Florian Eckert <fe@dev.tdt.de>
Date: Fri, 19 Nov 2021 09:51:02 +0100
Subject: [PATCH] firewall3: fix locking issue

By calling the command 'fw3 reload' several times at the same time, I
noticed that the locking was not working properly. It happened from time
to time that some firewall rules were present twice in the system!

By removing the 'unlink' systemcall, this error no longer occurred on my
systems.

Since fw3 does not run as a service, it makes no sense to delete this
lock file every time a filehandler is no longer open on this lock file,
because fw3 binary is not running.

If fw3 does run as a service then we can remove this lock file on
service stop. But this is not the case for fw3.

Signed-off-by: Florian Eckert <fe@dev.tdt.de>
---
 utils.c | 1 -
 1 file changed, 1 deletion(-)

--- a/utils.c
+++ b/utils.c
@@ -411,7 +411,6 @@ fw3_unlock_path(int *fd, const char *loc
 		warn("Cannot release exclusive lock: %s", strerror(errno));
 
 	close(*fd);
-	unlink(FW3_LOCKFILE);
 
 	*fd = -1;
 }
